src/concrete/ml/deployment/fhe_client_server.py

```diff
@@ -357,97 +388,78 @@ def get_serialized_evaluation_keys(self) -> bytes:
         return self.client.evaluation_keys.serialize()
 
     def quantize_encrypt_serialize(
-        self, x: Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]
-    ) -> Union[bytes, Tuple[bytes, ...]]:
+        self, *x: Optional[numpy.ndarray]
+    ) -> Union[Optional[bytes], Tuple[Optional[bytes], ...]]:
         """Quantize, encrypt and serialize the values.
 
         Args:
-            x (Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]): the values to quantize,
-                encrypt and serialize
+            x (Optional[numpy.ndarray]): The values to quantize, encrypt and serialize.
 
         Returns:
-            Union[bytes, Tuple[bytes, ...]]: the quantized, encrypted and serialized values
+            Union[bytes, Tuple[bytes, ...]]: The quantized, encrypted and serialized values.
         """
 
-        x = to_tuple(x)
-
         # Quantize the values
-        quantized_x = self.model.quantize_input(*x)
-
-        quantized_x = to_tuple(quantized_x)
+        q_x = to_tuple(self.model.quantize_input(*x))
 
         # Encrypt the values
-        enc_qx = self.client.encrypt(*quantized_x)
-
-        enc_qx = to_tuple(enc_qx)
+        q_x_enc = to_tuple(self.client.encrypt(*q_x))
 
         # Serialize the encrypted values to be sent to the server
-        serialized_enc_qx = tuple(e.serialize() for e in enc_qx)
+        q_x_enc_serialized = serialize_encrypted_values(*q_x_enc)
 
-        # Return a single value if the original input was a single value
-        return serialized_enc_qx[0] if len(serialized_enc_qx) == 1 else serialized_enc_qx
+        return q_x_enc_serialized
 
+    # We should find a better name for `serialized_encrypted_quantized_result`
+    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4476
```

Comment:
Did you put the fixme in things we need to change for 2.0?

Reply:
yes I did !